students/kevin_t/Lesson7/parallel.py

```diff
@@ -0,0 +1,141 @@
+
+    my_queue.put((customer_count, initial_count, final_count, local_end_time - local_start_time))
+
+def import_products(directory_name, customer_file, my_queue):
+    """Brings in products csv files, counts lines and adds to database"""
+    local_start_time = time.time()
+    products_count = 0
+
+    mongo = MongoDBConnection()
+
+    with mongo:
+        database = mongo.connection.store
+        products = database["products"]
+        products.drop()
+
+        initial_count = database.products.count_documents({})
+
+        with open(os.path.join(directory_name, customer_file)) as csvfile:
+            products_reader = csv.reader(csvfile, delimiter=',')
+            for row in products_reader:
+                products_count += 1
+                products_info = {"product_id": row[0], "description": row[1],
+                                 "product_type": row[2], "quantity_available": row[3]}
+                products.insert_one(products_info)
+
+        final_count = database.customer.count_documents({})
+
+    local_end_time = time.time()
+
+    my_queue.put((products_count, initial_count, final_count, local_end_time - local_start_time))
+
+def import_rentals(directory_name, customer_file, my_queue):
+    """Brings in rental csv files, counts lines and adds to database"""
+    local_start_time = time.time()
+    rentals_count = 0
+
+    mongo = MongoDBConnection()
+
+    with mongo:
+        database = mongo.connection.store
+        rentals = database["rentals"]
+        rentals.drop()
+
+        initial_count = database.rentals.count_documents({})
+
+        with open(os.path.join(directory_name, customer_file)) as csvfile:
+            rentals_reader = csv.reader(csvfile, delimiter=',')
+            for row in rentals_reader:
+                rentals_count += 1
+                rentals_info = {"product_id": row[0], "user_id": row[1], "rental_date": row[2],
+                                "return_date": row[3]}
+                rentals.insert_one(rentals_info)
+
+        final_count = database.customer.count_documents({})
+
+    local_end_time = time.time()
+
+    my_queue.put((rentals_count, initial_count, final_count, local_end_time - local_start_time))
+
+if __name__ == '__main__':
+    TOTAL_START_TIME = time.time()
+    MY_QUEUE = queue.Queue()
+
+    THREAD1 = threading.Thread(target=import_customers, args=('input_files', 'customers_l7.csv',
+                                                              MY_QUEUE))
+    THREAD1.start()
+    THREAD2 = threading.Thread(target=import_products, args=('input_files', 'products_l7.csv',
+                                                             MY_QUEUE))
+    THREAD2.start()
+    THREAD3 = threading.Thread(target=import_rentals, args=('input_files', 'rentals_l7.csv',
+                                                            MY_QUEUE))
+    THREAD3.start()
+
+    THREAD1.join()
+    THREAD2.join()
+    THREAD3.join()
+
+    CUSTOMERS = MY_QUEUE.get()
```

Comment:
There's a problem with this approach for getting results out of the queue, as not all threads will always return in the _correct_ order, so you might end up getting the _products_ results when trying to get _customers_. You can modify your results so that there's a field or something else that will indicate which thread they came for.